packages/core/src/electron-main/electron-application.ts

```diff
@@ -0,0 +1,466 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, named } from 'inversify';
+import { session, screen, globalShortcut, app, BrowserWindow, BrowserWindowConstructorOptions, Event as ElectronEvent, dialog } from 'electron';
+import * as path from 'path';
+import { Argv } from 'yargs';
+import { AddressInfo } from 'net';
+import { promises as fs } from 'fs';
+import { fork, ForkOptions } from 'child_process';
+import { FrontendApplicationConfig } from '@theia/application-package/lib/application-props';
+import URI from '../common/uri';
+import { FileUri } from '../node/file-uri';
+import { Deferred } from '../common/promise-util';
+import { MaybePromise } from '../common/types';
+import { ContributionProvider } from '../common/contribution-provider';
+import { ElectronSecurityToken } from '../electron-common/electron-token';
+const Storage = require('electron-store');
+const createYargs: (argv?: string[], cwd?: string) => Argv = require('yargs/yargs');
+
+/**
+ * Options passed to the main/default command handler.
+ */
+export interface MainCommandOptions {
```

Comment:
please qualify types to avoid polluting workspace symbols with duplicate names: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#unique-names